compose/service.py

```diff
@@ -798,6 +801,13 @@ def _get_container_host_config(self, override_options, one_off=False):
             init_path = options.get('init')
             options['init'] = True
 
+        nano_cpus = None
+        if 'cpus' in options:
+            nano_cpus = options.get('cpus') * 1000000000
+            if isinstance(nano_cpus, float) and not nano_cpus.is_integer():
```

Comment:
I'd really like us to do that check in `validation.py` so it can be caught when running `docker-compose config`